lib/jwt_signed_request.rb

```
@@ -77,7 +77,11 @@ def self.request_body(request:)
   private_class_method :request_body
 
   def self.verified_headers?(request:, claims:)
-    parsed_headers = JSON.parse(claims['headers'])
+    parsed_headers = begin
+      JSON.parse(claims['headers'])
+    rescue JSON::ParserError
+      {}
```

I would really like to understand what a _valid_ use case for when you expect there not to be any headers in the claim would look like. If there isn't one, I'm somewhat skeptical to the idea of silently swallowing this error.

---

Previously you could have passed an empty JSON hash and it would have passed the header verification. This simply allows you to pass nothing.

Taking another look I haven't properly handled the `nil` case, I'll be pushing another commit to handle that properly.

We could raise more specific errors if certain claims are missing but I would guess that could be giving more information to attackers.

---

After some offline discussion I'm happy to proceed as-is. I'm happy to accept this is not a regression, though I'd like to discuss whether we should allow empty/no headers in a separate issue.